mmdet3d/models/losses/paconv_regularization_loss.py

```diff
@@ -0,0 +1,106 @@
+import torch
+from torch import nn as nn
+
+from mmdet3d.ops import PAConv, PAConvCUDA
+from mmdet.models.builder import LOSSES
+from mmdet.models.losses.utils import weight_reduce_loss
+
+
+def weight_correlation(conv):
+    """Calculate correlations between kernel weights in Conv's weight bank as
+    regularization loss. The cosine similarity is used as metrics.
+
+    Args:
+        conv (nn.Module): A Conv modules to be regularized.
+            Currently we only support `PAConv` and `PAConvCUDA`.
+
+    Returns:
+        torch.Tensor: Correlations between each kernel weights in weight bank.
+    """
+    assert isinstance(conv, (PAConv, PAConvCUDA))
```

Comment:
Only support PAConv? Add the corresponding assertion message

Reply:
No. This loss regularizes the weight kernels in both `PAConv` and `PAConvCUDA`. Basically, `PAConv` and `PAConvCUDA` are just two different implementations with almost the same in-out. `PAConvCUDA` uses custom CUDA op to accelerate computation and reduce GPU memory usage.

I will update the assertion info.

Reply:
Changed them in #783